### ISSUE TITLE
iTerm2 with split panes unusable

### ISSUE THREAD

**Issue by jlaamanen** [withfig/fig#528]
### Sanity checks

- [X] I have searched [github.com/withfig/fig/issues](https://github.com/withfig/fig/issues?q=) and there are no duplicates of my issue

### Description:
I have iTerm2 installed (using Fish shell & Starship) and I use extensively iTerm's split panes. Listed some problems that make Fig unusable with my setup:
* Fig autocomplete always only works in the **last** created split pane
* Occasionally when creating a new split pane, iTerm2 ends the session with message:
  > **Warning**
  >
  > A session ended very soon after starting. Check that the command in profile "_[profile-name]_" is correct.
* A keystroke in a previous split pane causes all later created split panes to crash

Interestingly, the iTerm session crash occurs only when creating multiple split panes to the same direction - when altering between horizontal and vertical split panes it didn't happen, but the split panes would still crash with a keystroke in any of the earlier created split panes.

The problems remained even after quitting Fig - only after uninstalling Fig the problems disappear. Also tried uninstalling Starship, but it had no effect.

### Environment

```bash
Version 1.0.51 (B285) [Finnish]
UserShell: /usr/local/bin/fish
Bundle path: /Applications/Fig.app
Autocomplete: true
Settings.json: true
CLI installed: true
CLI tool path: /Users/joona/.fig/bin/fig
Accessibility: true
Number of specs: 1
SSH Integration: false
Tmux Integration: false
Keybindings path: <none>
iTerm Integration: true [Authenticated]
Hyper Integration: false
VSCode Integration: true
Docker Integration: true
Symlinked dotfiles: false
Only insert on tab: false
Installation Script: true
PseudoTerminal Path: /usr/local/Caskroom/google-cloud-sdk/latest/google-cloud-sdk/bin:/Users/joona/.fig/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin:/Users/joona/.npm-packages/bin
SecureKeyboardInput: false
SecureKeyboardProcess: <none>
Current active process: /usr/local/Cellar/fish/3.1.2/bin/fish (36005) - ttys012
Current working directory: /Users/joona
Current window identifier: 38061/17979614-9560-4882-8476-1BAEC99A70B7% (com.googlecode.iterm2)
PATH: /usr/local/Caskroom/google-cloud-sdk/latest/google-cloud-sdk/bin:/Users/joona/.fig/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin:/Users/joona/.npm-packages/bin
FIG_INTEGRATION_VERSION: 4
```

**Comment by jlaamanen** [comment]
Hi all, seems @mschrage pushed a fix for this: https://github.com/withfig/fig/issues/548#issuecomment-954301543

I ran `fig settings app.beta true` and the problem seems to have been disappeared. :+1: Do you want to keep this issue open until the fix gets out of beta? If not, feel free to close this issue.

**Comment by jeremybarbet** [comment]
I believed #548 wasn't the same issue, my fish sessions kept crashing before the fix.

However, regarding this one, if I open my default window arrangement, that has 6 panes, fig will only work in one of them. When I open new pane/tab it works as intended. Looks to be only an issue at the first initialisation. It happened to work and initialise all of them, but most of the time, there is only one initialised. Also trying to `fig source` within the pane doesn't work either.

Workaround from @brumm is working, but is quite dirty 😄

**Comment by mschrage** [comment]
@jeremybarbet it is possible that opening a bunch of sessions simultaneously causes issues when linking. Can you share your iTerm configuration here?

**Comment by jeremybarbet** [comment]
Sorry to ask, how do I do that?

**Comment by mschrage** [comment]
Go to iTerm preferences > Profiles, then select your profile from the list and click the "other actions" button at the bottom. (You can see it in the screenshot that @brumm shared.)

Also if you're open to it, I would love to jump on a debugging call with you next week! 😊 Here is my scheduling link: https://calendly.com/mschrage/30min

**Comment by clementloridan** [comment]
This is mine.  I will love to have an update of this bug, without this I can't use fig, split so important for me :'(

```
{
  "Working Directory" : "\/Users\/clementloridan",
  "Prompt Before Closing 2" : false,
  "Selected Text Color" : {
    "Red Component" : 0.76029282808303833,
    "Color Space" : "sRGB",
    "Blue Component" : 0.76030111312866211,
    "Alpha Component" : 1,
    "Green Component" : 0.7603108286857605
  },
  "Rows" : 25,
  "Ansi 11 Color" : {
    "Red Component" : 0.94509822130203247,
    "Color Space" : "sRGB",
    "Blue Component" : 0,
    "Alpha Component" : 1,
    "Green Component" : 0.81568628549575806
  },
  "Use Italic Font" : true,
  "Foreground Color" : {
    "Red Component" : 0.99999600648880005,
    "Color Space" : "sRGB",
    "Blue Component" : 1,
    "Alpha Component" : 1,
    "Green Component" : 1
  },
  "Right Option Key Sends" : 0,
  "Character Encoding" : 4,
  "Selection Color" : {
    "Red Component" : 0.70196080207824707,
    "Color Space" : "sRGB",
    "Blue Component" : 1,
    "Alpha Component" : 1,
    "Green Component" : 0.38431373238563538
  },
  "Mouse Reporting" : true,
  "Ansi 4 Color" : {
    "Red Component" : 0.4117647111415863,
    "Color Space" : "sRGB",
    "Blue Component" : 1,
    "Alpha Component" : 1,
    "Green Component" : 0.26274511218070984
  },
  "Non-ASCII Anti Aliased" : true,
  "Sync Title" : false,
  "Disable Window Resizing" : true,
  "Description" : "Default",
  "Close Sessions On End" : true,
  "Jobs to Ignore" : [
    "rlogin",
    "ssh",
    "slogin",
    "telnet"
  ],
  "Scrollback With Status Bar" : true,
  "Scrollback Lines" : 0,
  "Flashing Bell" : false,
  "Cursor Guide Color" : {
    "Red Component" : 0.70213186740875244,
    "Color Space" : "sRGB",
    "Blue Component" : 1,
    "Alpha Component" : 0.25,
    "Green Component" : 0.9268307089805603
  },
  "BM Growl" : true,
  "AWDS Window Directory" : "",
  "Ansi 3 Color" : {
    "Red Component" : 1,
    "Color Space" : "sRGB",
    "Blue Component" : 0,
    "Alpha Component" : 1,
    "Green Component" : 0.90588241815567017
  },
  "Use Non-ASCII Font" : false,
  "Link Color" : {
    "Red Component" : 0,
    "Color Space" : "sRGB",
    "Blue Component" : 0.73423302173614502,
    "Alpha Component" : 1,
    "Green Component" : 0.35916060209274292
  },
  "Shortcut" : "",
  "Background Image Location" : "",
  "Bold Color" : {
    "Red Component" : 0.97516018152236938,
    "Color Space" : "sRGB",
    "Blue Component" : 1,
    "Alpha Component" : 1,
    "Green Component" : 0.99032300710678101
  },
  "Unlimited Scrollback" : true,
  "Custom Command" : "Yes",
  "AWDS Tab Option" : "No",
  "Keyboard Map" : {
    "0xf728-0x80000" : {
      "Text" : "d",
      "Action" : 10
    },
    "0xf702-0x280000" : {
      "Text" : "b",
      "Action" : 10
    },
    "0xf70d-0x20000" : {
      "Text" : "[21;2~",
      "Action" : 10
    },
    "0x7f-0x80000" : {
      "Text" : "0x1b 0x7f",
      "Action" : 11
    },
    "0xf708-0x20000" : {
      "Text" : "[15;2~",
      "Action" : 10
    },
    "0x33-0x40000" : {
      "Text" : "0x1b",
      "Action" : 11
    },
    "0xf703-0x260000" : {
      "Text" : "[1;6C",
      "Action" : 10
    },
    "0xf729-0x20000" : {
      "Text" : "[1;2H",
      "Action" : 10
    },
    "0xf702-0x260000" : {
      "Text" : "[1;6D",
      "Action" : 10
    },
    "0x38-0x40000" : {
      "Text" : "0x7f",
      "Action" : 11
    },
    "0xf72b-0x40000" : {
      "Text" : "[1;5F",
      "Action" : 10
    },
    "0xf70c-0x20000" : {
      "Text" : "[20;2~",
      "Action" : 10
    },
    "0xf701-0x260000" : {
      "Text" : "[1;6B",
      "Action" : 10
    },
    "0x32-0x40000" : {
      "Text" : "0x00",
      "Action" : 11
    },
    "0xf707-0x20000" : {
      "Text" : "[1;2S",
      "Action" : 10
    },
    "0xf703-0x240000" : {
      "Text" : "[1;5C",
      "Action" : 10
    },
    "0xf700-0x260000" : {
      "Text" : "[1;6A",
      "Action" : 10
    },
    "0xf702-0x240000" : {
      "Text" : "[1;5D",
      "Action" : 10
    },
    "0x37-0x40000" : {
      "Text" : "0x1f",
      "Action" : 11
    },
    "0x3-0x200000" : {
      "Text" : "0xd",
      "Action" : 11
    },
    "0xf701-0x240000" : {
      "Text" : "[1;5B",
      "Action" : 10
    },
    "0xf70b-0x20000" : {
      "Text" : "[19;2~",
      "Action" : 10
    },
    "0xf703-0x220000" : {
      "Text" : "[1;2C",
      "Action" : 10
    },
    "0xf739-0x0" : {
      "Text" : "",
      "Action" : 13
    },
    "0xf728-0x0" : {
      "Text" : "0x4",
      "Action" : 11
    },
    "0xf706-0x20000" : {
      "Text" : "[1;2R",
      "Action" : 10
    },
    "0xf700-0x240000" : {
      "Text" : "[1;5A",
      "Action" : 10
    },
    "0xf702-0x220000" : {
      "Text" : "[1;2D",
      "Action" : 10
    },
    "0x36-0x40000" : {
      "Text" : "0x1e",
      "Action" : 11
    },
    "0xf70f-0x20000" : {
      "Text" : "[24;2~",
      "Action" : 10
    },
    "0xf701-0x220000" : {
      "Text" : "[1;2B",
      "Action" : 10
    },
    "0xf70a-0x20000" : {
      "Text" : "[18;2~",
      "Action" : 10
    },
    "0xf729-0x40000" : {
      "Text" : "[1;5H",
      "Action" : 10
    },
    "0xf72b-0x20000" : {
      "Text" : "[1;2F",
      "Action" : 10
    },
    "0xf700-0x220000" : {
      "Text" : "[1;2A",
      "Action" : 10
    },
    "0x30-0x200000" : {
      "Text" : "0",
      "Action" : 12
    },
    "0xf705-0x20000" : {
      "Text" : "[1;2Q",
      "Action" : 10
    },
    "0x31-0x200000" : {
      "Text" : "1",
      "Action" : 12
    },
    "0x33-0x200000" : {
      "Text" : "3",
      "Action" : 12
    },
    "0x35-0x40000" : {
      "Text" : "0x1d",
      "Action" : 11
    },
    "0x34-0x200000" : {
      "Text" : "4",
      "Action" : 12
    },
    "0x32-0x200000" : {
      "Text" : "2",
      "Action" : 12
    },
    "0x35-0x200000" : {
      "Text" : "5",
      "Action" : 12
    },
    "0x36-0x200000" : {
      "Text" : "6",
      "Action" : 12
    },
    "0x37-0x200000" : {
      "Text" : "7",
      "Action" : 12
    },
    "0x2d-0x40000" : {
      "Text" : "0x1f",
      "Action" : 11
    },
    "0x38-0x200000" : {
      "Text" : "8",
      "Action" : 12
    },
    "0x2a-0x200000" : {
      "Text" : "*",
      "Action" : 12
    },
    "0x39-0x200000" : {
      "Text" : "9",
      "Action" : 12
    },
    "0x2b-0x200000" : {
      "Text" : "+",
      "Action" : 12
    },
    "0xf70e-0x20000" : {
      "Text" : "[23;2~",
      "Action" : 10
    },
    "0x2d-0x200000" : {
      "Text" : "-",
      "Action" : 12
    },
    "0x2e-0x200000" : {
      "Text" : ".",
      "Action" : 12
    },
    "0xf709-0x20000" : {
      "Text" : "[17;2~",
      "Action" : 10
    },
    "0x7f-0x100000" : {
      "Text" : "0x15",
      "Action" : 11
    },
    "0x2f-0x200000" : {
      "Text" : "\/",
      "Action" : 12
    },
    "0xf704-0x20000" : {
      "Text" : "[1;2P",
      "Action" : 10
    },
    "0x34-0x40000" : {
      "Text" : "0x1c",
      "Action" : 11
    },
    "0xf703-0x280000" : {
      "Text" : "f",
      "Action" : 10
    }
  },
  "Ansi 14 Color" : {
    "Red Component" : 0.47516816854476929,
    "Color Space" : "sRGB",
    "Blue Component" : 0.98378199338912964,
    "Alpha Component" : 1,
    "Green Component" : 0.90928202867507935
  },
  "Ansi 2 Color" : {
    "Red Component" : 0.22745098173618317,
    "Color Space" : "sRGB",
    "Blue Component" : 0,
    "Alpha Component" : 1,
    "Green Component" : 0.85098040103912354
  },
  "Background Image Is Tiled" : false,
  "Send Code When Idle" : false,
  "ASCII Anti Aliased" : true,
  "Tags" : [

  ],
  "Ansi 9 Color" : {
    "Red Component" : 0.97647064924240112,
    "Color Space" : "sRGB",
    "Blue Component" : 0.10980390757322311,
    "Alpha Component" : 1,
    "Green Component" : 0.16470575332641602
  },
  "Use Bold Font" : true,
  "Silence Bell" : false,
  "Ansi 12 Color" : {
    "Red Component" : 0.40937519073486328,
    "Color Space" : "sRGB",
    "Blue Component" : 1,
    "Alpha Component" : 1,
    "Green Component" : 0.44485551118850708
  },
  "Window Type" : 0,
  "Use Bright Bold" : true,
  "Cursor Text Color" : {
    "Red Component" : 0.99718040227890015,
    "Color Space" : "sRGB",
    "Blue Component" : 0.95852619409561157,
    "Alpha Component" : 1,
    "Green Component" : 0.99865502119064331
  },
  "Default Bookmark" : "No",
  "Cursor Color" : {
    "Red Component" : 0.98039215803146362,
    "Color Space" : "sRGB",
    "Blue Component" : 0,
    "Alpha Component" : 1,
    "Green Component" : 0.81568628549575806
  },
  "Ansi 1 Color" : {
    "Red Component" : 0.85098040103912354,
    "Color Space" : "sRGB",
    "Blue Component" : 0.16078431904315948,
    "Alpha Component" : 1,
    "Green Component" : 0.015686275437474251
  },
  "Name" : "Default",
  "Blinking Cursor" : true,
  "Guid" : "98A1CA2E-080E-4A5F-A717-0EA10FF63EB6",
  "Idle Code" : 0,
  "Ansi 10 Color" : {
    "Red Component" : 0.26274508237838745,
    "Color Space" : "sRGB",
    "Blue Component" : 0.14901953935623169,
    "Alpha Component" : 1,
    "Green Component" : 0.83137261867523193
  },
  "Ansi 8 Color" : {
    "Red Component" : 0.40781760215759277,
    "Color Space" : "sRGB",
    "Blue Component" : 0.4078223705291748,
    "Alpha Component" : 1,
    "Green Component" : 0.40782788395881653
  },
  "Badge Color" : {
    "Red Component" : 1,
    "Color Space" : "sRGB",
    "Blue Component" : 0,
    "Alpha Component" : 0.5,
    "Green Component" : 0.1491314172744751
  },
  "Ambiguous Double Width" : false,
  "Blur Radius" : 12.010737230329946,
  "Cursor Type" : 1,
  "Ansi 0 Color" : {
    "Red Component" : 0,
    "Color Space" : "sRGB",
    "Blue Component" : 0,
    "Alpha Component" : 1,
    "Green Component" : 0
  },
  "AWDS Pane Directory" : "",
  "Blur" : true,
  "Normal Font" : "MesloLGS-NF-Regular 12",
  "Vertical Spacing" : 1,
  "Use Underline Color" : false,
  "Ansi 7 Color" : {
    "Red Component" : 0.7810397744178772,
    "Color Space" : "sRGB",
    "Blue Component" : 0.78104829788208008,
    "Alpha Component" : 1,
    "Green Component" : 0.78105825185775757
  },
  "AWDS Tab Directory" : "",
  "AWDS Pane Option" : "No",
  "Command" : "\/bin\/zsh",
  "Terminal Type" : "xterm-256color",
  "Horizontal Spacing" : 1,
  "Option Key Sends" : 0,
  "Only The Default BG Color Uses Transparency" : false,
  "Ansi 15 Color" : {
    "Red Component" : 0.99999600648880005,
    "Color Space" : "sRGB",
    "Blue Component" : 1,
    "Alpha Component" : 1,
    "Green Component" : 1
  },
  "Ansi 6 Color" : {
    "Red Component" : 0,
    "Color Space" : "sRGB",
    "Blue Component" : 0.78166204690933228,
    "Alpha Component" : 1,
    "Green Component" : 0.77425903081893921
  },
  "Transparency" : 0.036652649235604379,
  "Initial Text" : "fish",
  "Background Color" : {
    "Red Component" : 0.11764705926179886,
    "Color Space" : "sRGB",
    "Blue Component" : 0.25098040699958801,
    "Alpha Component" : 1,
    "Green Component" : 0.11372549086809158
  },
  "Screen" : -1,
  "AWDS Window Option" : "No",
  "Non Ascii Font" : "Monaco 12",
  "Ansi 13 Color" : {
    "Red Component" : 1,
    "Color Space" : "sRGB",
    "Blue Component" : 1,
    "Alpha Component" : 1,
    "Green Component" : 0.46518981456756592
  },
  "Use Tab Color" : true,
  "Columns" : 80,
  "Visual Bell" : true,
  "Custom Directory" : "Recycle",
  "Ansi 5 Color" : {
    "Red Component" : 1,
    "Color Space" : "sRGB",
    "Blue Component" : 0.43921792507171631,
    "Alpha Component" : 1,
    "Green Component" : 0.17081758379936218
  }
}
```

**Comment by mschrage** [comment]
Hey @clementloridan, it doesn't seem like your profile automatically opens a splitscreen window? Can you share a screen recording of your issue?
